haystack/document_store/milvus.py

```diff
@@ -403,9 +403,8 @@ def delete_documents(self, index: Optional[str] = None, filters: Optional[Dict[s
         if status.code != Status.SUCCESS:
             raise RuntimeError(f'Milvus has collection check failed: {status}')
         if ok:
-            status = self.milvus_server.drop_collection(collection_name=index)
-            if status.code != Status.SUCCESS:
-                raise RuntimeError(f'Milvus drop collection failed: {status}')
+            existing_docs = super().get_all_documents(filters=filters, index=index)
```

Comment:
For large indices, this might run out of memory. For now, I think it's okay if we put an if-else block here that uses the current `drop_collection` if filters are None and your code if filters are present. 
If we experience memory issues, we might add a method `get_vector_ids(filters, index)` to the SQLDocumentStore that only returns the needed ids and not the texts of docs etc.

Reply:
Thanks for the suggestion! I'll amend the PR to reflect the changes.